docs/developer-guides/merkle-root-construction.md

```diff
@@ -2,47 +2,46 @@
-     for A|B concatenation demonstrated below:
-         sha256(sha256("82501c1178fa0b222c1f3d474ec726b832013f0a532b44bb620cce8624a5feb1169e1e83e930853391bc6f35f605c6754cfead57cf8387639d3b4096c54f18f4".decode("hex")).digest()).digest().encode("hex_codec")
--->
-
-![Example Merkle Tree Construction](/img/dev/en-merkle-tree-construction.svg)
-
-TXIDs and intermediate hashes are always in internal byte order when they're
-concatenated, and the resulting merkle root is also in internal byte
-order when it's placed in the block header.
+This page explains what merkle trees are, how Decred uses them and provides step-by-step instructions for constructing compliant Merkle trees.
+
+### Merkle Trees
+
+
+A merkle tree is a data structure that can be used to store hashes of arbitrary data. In the context of Decred, this might be transaction or ticket data. 
+
+Merkle trees are created by repeatedly hashing pairs of nodes until there is only one hash left, the Merkle Root. They are constructed from the bottom up, from hashes of the leaf nodes (hashes of the raw input data). The diagram below shows a merkle root constructed from eight leaf nodes. 
+
+![Merkle Tree](/img/core-blockchain-concepts/merkle_tree.svg)
+
+Merkle trees are particularly useful for blockchains, as they efficiently and compactly summarize larger data sets.
+
+
+### Merkle Trees in Decred
+
+Merkle roots are used to create compact summaries of all transactions in a block. These merkle roots can then be queried to quickly and efficiently verify whether or not a a transaction is included in a block. 
+
+The block header of each block contains the merkle roots of two merkle trees, the `MerkleRoot` tree and `StakeRoot` tree. The `MerkleRoot` tree contains hashes of all regular transaction IDs. The `StakeRoot` tree contains hashes of stake-based transaction IDs (ticket purchases, votes, revocations, etc.). 
+
+#### Hash Function
+
+Unless otherwise specified, all hashing operations **must** be performed with the BLAKE-256 hash function with 14 rounds.
+
+#### Merkle Root Construction
+
+All Merkle trees **must** be constructed according to the unique Merkle tree construct originally implemented by Satoshi, which relies on construction of the tree from an ordered list as follows:
+
+1. Calculate list of BLAKE-256 hashes of original data (e.g. transaction IDs) 
+1. If the list is empty, return a 32-byte root hash of all zeroes; otherwise
+1. While the list contains two or more items:
+	- If the number of items remaining in the list is odd, duplicate the final hash
+	- Combine each pair of adjacent entries with the BLAKE-256 hash of the two entries concatenated together. The list will have ceil(N/2) entries remaining after combining the adjacent entries
+1. Return the final remaining item in the list as the Merkle root
+
+The below diagram illustrates these steps.
+
+![Merkle Tree Calculation](/img/core-blockchain-concepts/merkle_root_calc.svg)
+
+!!! warning "Warning"
```

Comment:
I think this warning should really be moved above the diagram. It feels strange that the last element on the page is a warning box, ie. a high priority message. It should be more prominent imo

Reply:
Good call. Moved.